docs/examples/use_cases/pytorch/resnet50/main.py

```diff
@@ -431,10 +553,20 @@ def validate(val_loader, model, criterion):
 
     end = time.time()
 
-    for i, data in enumerate(val_loader):
-        input = data[0]["data"]
-        target = data[0]["label"].squeeze(-1).long()
-        val_loader_len = int(val_loader._size / args.batch_size)
+    if args.disable_dali:
+        data_iterator = data_prefetcher(val_loader)
+        data_iterator = iter(data_iterator)
+    else:
+        data_iterator = val_loader
+
+    for i, data in enumerate(data_iterator):
+        if args.disable_dali:
+            input, target = data
+            val_loader_len = len(val_loader)
+        else:
+            input = data[0]["data"]
+            target = data[0]["label"].squeeze(-1).long()
+            val_loader_len = int(math.ceil(data_iterator._size / args.batch_size))
```

Comment:
Super nitpick, but I'd change the for and if/else order as the args.disable_dali does not change throughout iteration.

Reply:
(Actually you could hoist just the corresponding branches to the if above, right?)

Reply:
It doesn't, we I can at most extract `val_loader_len` computation from the loop, still it need to stay inside as the loop loops over the `data_iterator` that produces `data` that needs a different handling depending on the condition.

Reply:
Oh, there's obviously a whole lot of code below, I just didn't notice it, sorry for bothering. :D

Reply:
No problem.